### PR TITLE
Update py-bokeh: new version & dependencies"

### PR DESCRIPTION
**Pull request by brietzke**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-bokeh/package.py [hunk 1]
```diff
@@ -12,15 +12,18 @@ class PyBokeh(PythonPackage):
...
 
+    version('1.3.4', 'e2d97bed5b199a10686486001fed5c854e4c04ebe28859923f27c52b93904754')
     version('0.12.2', '2d1621bffe6e2ab9d42efbf733861c4f')
 
-    depends_on('python@2.6:')
-    depends_on('py-six@1.5.2:',       type=('build', 'run'))
-    depends_on('py-requests@1.2.3:',  type=('build', 'run'))
-    depends_on('py-pyyaml@3.10:',     type=('build', 'run'))
+    depends_on('python@2.6:', when='@0.12.2')
+    depends_on('python@2.7:', when='@1.3.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-bokeh/package.py]: Probably `type=('build', 'run')`

Comment (brietzke) [reply var/spack/repos/builtin/packages/py-bokeh/package.py]: I dont know.
The default for Class PythonPackage for its python-dependency seems to be build, link, run
and it has been unspecfied for py-bokeh before

Comment (brietzke) [reply var/spack/repos/builtin/packages/py-bokeh/package.py]: thanks for answering this within #12725

**Comment by adamjstewart** [comment]
Can you resolve the merge conflicts?

**Pull request merged**